applylib/applyset/applyset.go

```diff
@@ -110,11 +188,23 @@ func (a *ApplySet) ApplyOnce(ctx context.Context) (*ApplyResults, error) {
 		gvk := obj.GroupVersionKind()
 		nn := types.NamespacedName{Namespace: ns, Name: name}
 
-		restMapping, err := a.restMapper.RESTMapping(gvk.GroupKind(), gvk.Version)
-		if err != nil {
-			results.applyError(gvk, nn, fmt.Errorf("error getting rest mapping for %v: %w", gvk, err))
+		restMappingResult := restMappings[gvk]
+		if restMappingResult.err != nil {
+			results.applyError(gvk, nn, fmt.Errorf("error getting rest mapping for %v: %w", gvk, restMappingResult.err))
 			continue
 		}
+
+		restMapping := restMappingResult.restMapping
+		if restMapping == nil {
+			// Should be impossible
+			results.applyError(gvk, nn, fmt.Errorf("rest mapping result not found for %v", gvk))
+		}
+
+		if err := a.updateManifestLabel(obj, kapplyset.LabelsForMember()); err != nil {
+			klog.Errorf("unable to update label for %v/%v %v: %v", obj.GetName(), obj.GetNamespace(), gvk, err)
```

Comment:
`return fmt.Errorf` , I think, otherwise we will leak the object (?)

Reply:
Sure. I was thinking that updating the manifests labels is purely for pruning purpose, pruning may not even be enabled so ignoring the errors can at least guarantee all manifests are deployed.